### PR TITLE
Help wanted Issue search

### PR DESCRIPTION
**Pull request by royka**
https://github.com/librariesio/libraries.io/issues/471

### CODE REVIEW

File: app/models/concerns/issue_search.rb [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    include Elasticsearch::Model
+    include Elasticsearch::Model::Callbacks
+
+
+    FIELDS = ['full_title^2', 'exact_title^2', 'body']
+
+    settings index: { number_of_shards: 1, number_of_replicas: 0 } do
+      mapping do
+        indexes :full_title, :analyzer => 'snowball', :boost => 6
+        indexes :exact_title, :index => :not_analyzed, :boost => 2
```
Comment (andrew) [inline app/models/concerns/issue_search.rb]: No need for the exact title on this, it's just a hack for priority in search based on exact matching

File: app/models/concerns/issue_search.rb [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        indexes :number
+        indexes :state
+
+      end
+    end
+
+    after_save() { __elasticsearch__.index_document }
+
+    def as_indexed_json(options = {})
+      as_json methods: [:exact_title]
```
Comment (andrew) [inline app/models/concerns/issue_search.rb]: Need to add in `github_repository_language` and `github_repository_license` here and add the same named methods to `GitHubIssue`

File: app/models/concerns/issue_search.rb [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    end
+
+    after_save() { __elasticsearch__.index_document }
+
+    def as_indexed_json(options = {})
+      as_json methods: [:title, :github_contributions_count, :stars]
+    end
+
+    def exact_title
+      full_title
```
Comment (andrew) [inline app/models/concerns/issue_search.rb]: You can remove this

File: app/views/help_wanted/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<% title @title %>
+<div class='row'>
+  <div class='col-md-8'>
+      <%= render @github_issues %>
+      <%= will_paginate @github_issues %>
```
Comment (andrew) [inline app/views/help_wanted/index.html.erb]: Change this to `will_paginate @search`

File: config/routes.rb [hunk 5]
```diff
@@ -88,6 +88,8 @@
   get 'deprecated-libraries', to: 'projects#deprecated', as: :deprecated
   get 'removed-libraries', to: 'projects#removed', as: :removed
 
+  get '/issues/help-wanted', to: 'help_wanted#index', as: :help_wanted
+
```
Comment (andrew) [inline config/routes.rb]: Can you make this `/help-wanted`

File: app/models/concerns/issue_search.rb [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+        indexes :stargazers_count, type: 'integer'
+        indexes :github_id, type: 'integer'
+        indexes :github_contributions_count, type: 'integer'
+
+        indexes :github_repository_language
+        indexes :github_repository_license
+
+        indexes :number
+        indexes :labels
```
Comment (andrew) [inline app/models/concerns/issue_search.rb]: This should be `indexes :labels, :analyzer => 'keyword'`

**Comment by BPScott** [comment]
Rather than hard coding to the "help wanted" label it'd be neat if the issue search could allow facets so you could search for "ham" to give all issues that mention ham in the title/description or "label:'help wanted'" to search for for issues tagged with that label. We could even do "repo:librariesio/libraries.io" to search within a given repo.

**Comment by andrew** [comment]
Merging this, making tweaks and hopefully deploying it tonight ✊  

Thanks so much for working on this @royka!

**Pull request merged**